Ag code

bepeck opened the pull request.

agivdel committed:
add Searcher class and their methods for counting of files in the directory

agivdel committed:
write method for searching copies of file in the directory by Files.getLastModifiedTime()

agivdel committed:
delete FileAndTime class, use lambda for filter copied files

agivdel committed:
add validation for the directory of searching

agivdel committed:
change directorySelect() and run()

agivdel committed:
minor changes inside the method iterationFilesFrom()

agivdel committed:
create a new class Doubles, create the second method to search for duplicate files (with Double object)

agivdel committed:
extract both methods for searching the duplicates

agivdel committed:
change the name of the static method in Doubles class

bepeck commented on src/main/java/agivdel/copiedFilesSearch/Searcher.java:
```diff
@@ -0,0 +1,85 @@
+            fileMap.putIfAbsent(fileTime, new ArrayList<>());
+            fileNameList = fileMap.get(fileTime);
+            fileNameList.add(file.getName());
+            fileMap.put(fileTime, fileNameList);
```
Почему ты думаешь, что файлы, у которых fileTime одинаковый, являются дубликатами?

agivdel replied:
намек понятен... переносим (копируем) снаружи себе папку с файлами - и у нас появляется куча файлов с одинаковым fileTime.

bepeck commented on src/main/java/agivdel/copiedFilesSearch/Searcher.java:
```diff
@@ -0,0 +1,85 @@
+//сравнение имен - ненадежно
+//сравнение расширений - ненадежно
+//сравнение размеров - ненадежно
+//сравнение побитово - ненадежно
```
Почему?

agivdel replied:
Прочитал, что некоторые файлы при копировании могут терять в размере. Не знаю, насколько это актуально для современных файловых систем, но раньше такое было.

bepeck replied:
Давай сделаем допущение, что не могут.

agivdel committed:
find duplicate files by last modified time and file size

agivdel committed:
clear Main class

agivdel committed:
rewrite time duplicates searching with stream

agivdel committed:
add method for calculating checksum (by CRC32)

agivdel committed:
make searching of duplicates by checksum (CRC32) and last modified time

agivdel committed:
add two way of duplicates searching
(1: checksum CRC32 first, then last modified time;
2: last modified time first, then checksum)

agivdel committed:
extract methods;
uncorrected error with repeated call of the stream

agivdel committed:
correct Doubles class (delete field FileTime)

agivdel committed:
add method for transform Stream<List<File>> to Stream<Doubles>;
add methods for Doubles class: size(), isEmpty().

agivdel committed:
add method for transform Stream<List<File>> to Stream<Doubles>;

agivdel committed:
rewrite method getCRC32()

agivdel committed:
rewrite method toString for Doubles class

agivdel committed:
remove unnecessary methods (size(), isEmpty(), merge())

agivdel committed:
remove unnecessary methods

agivdel committed:
rewrite method input()

agivdel committed:
rewrite methods getTimeDoubles() and splitByChecksum()

agivdel committed:
rewrite method  iterationFilesFrom()

agivdel committed:
rewrite method splitByChecksum()

agivdel committed:
rewrite method getCRC32()

agivdel committed:
add all methods duplicates for work with stream

agivdel committed:
delete outer Doubles class;
use streams and lists in methods alternately

agivdel committed:
add sort by creation time of file in method printAllDoubles()

agivdel committed:
add method removeZeroSize()

agivdel committed:
add method sortByCreateTime()

agivdel committed:
delete method sortByCreateTime(),
add file sorting in Doubles class constructor

agivdel committed:
make outer Doubles class

agivdel committed:
add TUI class,
methods of handling user requests and iteration of the files moved to class TUI

agivdel committed:
rewrite method validateDir()

agivdel committed:
extract class Walker for iteration of files

agivdel committed:
transfer main method run() to Main class, class TUI - only for input and print

agivdel committed:
removal transfer main method run() to Main class

agivdel committed:
add tests for two methods of Walker class

agivdel committed:
rewrite names of test methods of Walker class

agivdel committed:
add SearcherTest class with one test method for only public method of Searcher class

agivdel committed:
add comments for WalkerTest class

agivdel committed:
add comments for SearcherTest class

agivdel committed:
rewrite the call of test resources directory in all tests

agivdel committed:
rewrite all test methods

agivdel committed:
add test methods and javadoc for it

agivdel committed:
add test methods

agivdel committed:
delete old import

agivdel committed:
add test methods for input`s validation

agivdel committed:
change printAllDoubles() method

agivdel committed:
rewrite all tests of TUITest

agivdel committed:
rewrite all tests of WalkerTest

agivdel committed:
rewrite all tests of TUITest

agivdel committed:
rewrite all tests of SearcherTest

agivdel committed:
change argument`s name in Doubles constructor

agivdel committed:
delete sorting by creation time in Doubles constructor

agivdel committed:
add method getDoublesByChecksumThenTime() in Searcher class

agivdel committed:
add method getDoublesByChecksumThenTime() in Searcher class

agivdel committed:
add the option to select search by checksum or by last modified time in Searcher class

agivdel committed:
change text for console output

agivdel committed:
delete old import

agivdel committed:
rewrite method printAllDoubles()

agivdel committed:
rewrite some methods

agivdel committed:
fix before changing

Commit:
handle console commands framework

Commit:
handle console commands framework - example

agivdel committed:
add convolution of the framework

agivdel committed:
make slightly shorter

agivdel committed:
rename branch;
delete ag code
